src/schedlib/policies/sat.py

```diff
@@ -154,62 +251,184 @@
     "",
     "    pid.acq.start()",
     "",
-]
+    ]
 
-wrap_up = [
-    "# go home",
-    "run.acu.move_to(az=180, el=48)",
-    "",
-    "time.sleep(1)"
-]
-
-ufm_relock = [
-    "############# Daily Relock ######################",
-    "for smurf in pysmurfs:",
-    "    smurf.zero_biases.start()",
-    "for smurf in pysmurfs:",
-    "    smurf.zero_biases.wait()",
-    "",
-    "time.sleep(120)",
-    "run.smurf.take_noise(concurrent=True, tag='oper,take_noise,res_check')",
-    "",
-    "run.smurf.uxm_relock(concurrent=True)",
-    "#################################################", 
-]
+@cmd.operation(name='sat.ufm_relock', return_duration=True)
+def ufm_relock(state):
+    if state.last_ufm_relock is None:
+        doit = True
+    elif (state.curr_time - state.last_ufm_relock).total_seconds() > 12*u.hour:
+        doit = True
+    else:
+        doit = False
 
-hwp_spin_up = [
-    "############# Start HWP ######################",
-    "HWPPrep()",
-    "forward = True",
-    "hwp_freq = 2.0",
-    "HWPSpinUp()",
-]
+    if doit:
+        state = state.replace(last_ufm_relock=state.curr_time)
+        return state, 15*u.minute, [
+            "for smurf in pysmurfs:",
+            "    smurf.zero_biases.start()",
+            "for smurf in pysmurfs:",
+            "    smurf.zero_biases.wait()",
+            "",
+            "time.sleep(120)",
+            "run.smurf.take_noise(concurrent=True, tag='oper,take_noise,res_check')",
+            "run.smurf.uxm_relock(concurrent=True)",
+            "",
+        ]
+    else:
+        return state, 0, ["# no ufm relock needed at this time"]
+
+@cmd.operation(name='sat.hwp_spin_up', return_duration=True)
+def hwp_spin_up(state, disable_hwp=False):
+    if not disable_hwp and not state.hwp_spinning:
+        state = state.replace(hwp_spinning=True)
+        return state, 20*u.minute, [
+            "HWPPrep()",
+            "forward = True",
+            "hwp_freq = 2.0",
+            "HWPSpinUp()",
+        ]
+    return state, 0, ["# hwp disabled or already spinning"]
```

Comment:
The "hwp disabled or already spinning" get's into the comments generated in the schedule file. It's going to make me worried about the HWP state each time. Could you make these return "hwp already spinning" or "hwp disabled" depending on the actual reason for skipping? Same with spinning down.